Python/optimizer.c

```diff
@@ -394,6 +394,15 @@ executor_traverse(PyObject *o, visitproc visit, void *arg)
     return 0;
 }
 
+static int
+executor_is_gc(PyObject *o)
+{
+    if ((PyObject *)&COLD_EXITS[0] <= o && o < (PyObject *)&COLD_EXITS[COLD_EXIT_COUNT]) {
```

Comment:
This is undefined behavior in C. Although it will mostly work in practice.

Reply:
So I take it this should work?
```suggestion
    if (_Py_IsImmortal(o)) {
```

Reply:
Yes

Reply:
Maybe add a similar thing to the top of `executor_traverse()`? (Although, really, cold exits don't have exits, so their `exit_count` should always be zero, right? So why bother.

Anyway, see gh-118182 for a fix.